src/test/java/com/googleinterns/zoomtube/utils/CommentUtilTest.java

```diff
@@ -0,0 +1,96 @@
+import com.googleinterns.zoomtube.data.Comment;
+import java.io.IOException;
+import java.util.Date;
+import java.util.Optional;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public final class CommentUtilTest {
+  // Needed for accessing datastore services while creating an Entity.
+  private final LocalServiceTestHelper testServices = new LocalServiceTestHelper();
+
+  @Before
+  public void setUp() {
+    testServices.setUp();
+  }
+
+  @After
+  public void tearDown() {
+    testServices.tearDown();
+  }
+
+  @Test
+  public void fromEntity_shouldReturnCommentFromEntity() throws IOException {
+    Key lecture = KeyFactory.createKey(LectureUtil.KIND, 12345);
+    Key parent = KeyFactory.createKey(CommentUtil.KIND, 67890);
+    Date timestamp = new Date(123);
+    User author = new User("test@example.com", "example.com");
+    String content = "Test content";
+    Date dateNow = new Date();
+    Entity entity = new Entity(CommentUtil.KIND);
+    entity.setProperty(CommentUtil.LECTURE, lecture);
+    entity.setProperty(CommentUtil.PARENT, parent);
+    entity.setProperty(CommentUtil.TIMESTAMP, timestamp);
+    entity.setProperty(CommentUtil.AUTHOR, author);
+    entity.setProperty(CommentUtil.CONTENT, content);
+    entity.setProperty(CommentUtil.CREATED, dateNow);
+
+    Comment comment = CommentUtil.createComment(entity);
+
+    assertThat(comment.commentKey()).isEqualTo(entity.getKey());
+    assertThat(comment.lectureKey()).isEqualTo(lecture);
+    assertThat(comment.parentKey().isPresent()).isTrue();
+    assertThat(comment.parentKey().get()).isEqualTo(parent);
+    assertThat(comment.timestamp()).isEqualTo(timestamp);
+    assertThat(comment.author()).isEqualTo(author);
+    assertThat(comment.content()).isEqualTo(content);
+    assertThat(comment.created()).isEqualTo(dateNow);
+  }
+
+  @Test
+  public void createEntity_shouldReturnEntityWithProperties() throws IOException {
+    Key lectureKey = KeyFactory.createKey(LectureUtil.KIND, 12345);
+    Key parentKey = KeyFactory.createKey(CommentUtil.KIND, 67890);
+    Date timestamp = new Date(123);
+    User author = new User("test@example.com", "example.com");
+    String content = "Test content";
+    Date dateNow = new Date();
+
+    Entity entity = CommentUtil.createEntity(
+        lectureKey, Optional.of(parentKey), timestamp, author, content, dateNow);
+
+    assertThat(entity.getProperty(CommentUtil.LECTURE)).isEqualTo(lectureKey);
+    assertThat(entity.getProperty(CommentUtil.PARENT)).isEqualTo(parentKey);
+    assertThat(entity.getProperty(CommentUtil.TIMESTAMP)).isEqualTo(timestamp);
+    assertThat(entity.getProperty(CommentUtil.AUTHOR)).isEqualTo(author);
+    assertThat(entity.getProperty(CommentUtil.CONTENT)).isEqualTo(content);
+    assertThat(entity.getProperty(CommentUtil.CREATED)).isEqualTo(dateNow);
+  }
+}
```

Comment:
please add another unit test for  creating an entity with no parent as that is an edge case.

Reply:
Added